src/composer.ts

```diff
@@ -483,26 +578,27 @@ export class Composer<TContext extends Context>
    */
   static command<TContext extends Context>(
     command: MaybeArray<string>,
-    ...fns: NonemptyReadonlyArray<Middleware<TContext>>
-  ) {
+    ...fns: MatchedMiddleware<TContext, 'text'>
+  ): Middleware.Fn<TContext> {
     if (fns.length === 0) {
       // @ts-expect-error
       return Composer.entity(['bot_command'], command)
     }
     const commands = normalizeTextArguments(command, '/')
-    return Composer.mount(
+    return Composer.mount<TContext, 'text'>(
       'text',
-      Composer.lazy<TContext>((ctx) => {
+      Composer.lazy<TContext, MatchedContext<TContext, 'text'>>((ctx) => {
```

Comment:
Is the 2nd type param necessary in `Composer.lazy`? It seems to work without.
```suggestion
      Composer.lazy<MatchedContext<TContext, 'text'>>((ctx) => {
```

Reply:
Not for me.
```
Argument of type 'Fn<TContext>' is not assignable to parameter of type 'Middleware<MatchedContext<TContext, "text">>'.
  Type 'Fn<TContext>' is not assignable to type 'Fn<MatchedContext<TContext, "text">>'.
    Type 'MatchedContext<TContext, "text">' is not assignable to type 'TContext'.
      'MatchedContext<TContext, "text">' is assignable to the constraint of type 'TContext', but 'TContext' could be instantiated with a different subtype of constraint 'Context'.ts(2345)
```

Reply:
I just did `npm ci` and have no issue with the change, I'll push.

Reply:
Typechecking passed on CI. Now only tests of legacy `Composer.dispatch` need to be removed.